apps/evm_family/evm_helpers.c

```diff
@@ -223,12 +374,31 @@ bool evm_get_msg_data_hash(const evm_sign_msg_context_t *ctx, uint8_t *digest) {
           ctx->msg_data, ctx->init.total_msg_size, digest);
     } break;
 
-    case EVM_SIGN_MSG_TYPE_SIGN_TYPED_DATA:
-      // TODO: Prepare typed data message hash
-      break;
+    case EVM_SIGN_MSG_TYPE_SIGN_TYPED_DATA: {
+      result = evm_get_typed_struct_data_digest(&(ctx->typed_data), digest);
+    } break;
     default:
       break;
   }
 
   return result;
+}
+
+void evm_init_typed_data_display_node(
```

Comment:
let's keep this separate into `evm_typed_data.c` or something similar. it's not a generic helper.

Reply:
updated